scripts/deploy.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -e          # Exit with nonzero exit code if anything fails
+set -o verbose  # Print commands that are executed
+
+SOURCE_BRANCH=$CIRCLE_BRANCH
+TARGET_BRANCH="gh-pages"
+
+if [ "$CIRCLE_PULL_REQUEST" != "" ]; then
+  echo "Skipping deployment because this is a PR merge build."
+  exit 0
+fi
+
+if [ ! -d "dist" ]; then
+  echo "The dist/ directory doesn't exist; you must \`npm run build\` before deploying."
+  exit 1
+fi
+
+if [ ! -d "test-results" ]; then
+  echo "The test-results/ directory doesn't exist; you must \`npm run test\` before deploying."
+  exit 1
+fi
+
+if [ ! -d "test-results/coverage" ]; then
+  echo "The test-results/coverage/ directory doesn't exist; you must \`npm run test\` before deploying."
+  exit 1
+fi
+
+git config user.name > /dev/null || git config user.name 'Circle CI'
+git config user.email > /dev/null || git config user.email 'robots@circleci.com'
+
+# Move everything over to the gh-pages branch
+set +e
+git remote set-branches --add origin $TARGET_BRANCH
+git add --all
+git commit -m "Dummy commit"
+(git fetch origin $TARGET_BRANCH && git checkout -t origin/$TARGET_BRANCH) \
+  || git checkout --orphan $TARGET_BRANCH # In case the gh-pages branch didn't exist before
+set -e
+
+if [ ! -d "branches" ]; then
+  # Make the branches dir if it doesn't exist
+  mkdir branches
+fi
+
+# Rename the dist directory to match the source branch name
+mv dist branches/$SOURCE_BRANCH
+
+# Remove unneeded files from gh-pages
+shopt -s extglob
+rm -rf !(test-results|dist|assets|branches)
+
+if [ -f ".gitignore"]; then
+    rm .gitignore # remove our .gitignore so we can grab certain items
+fi
+
+# Move test-results output into source branch
+mv test-results branches/$SOURCE_BRANCH/unit-tests
```

Comment:
is 'unit-tests' just like standard or what?

Reply:
Nope, that's just the way it was in the old script, wondering if it was done to skirt around that .gitignore issue. Can rename to test-results? Test/coverage stuff will also get another pass when I start doing Coveralls work + work on hooking test results into CircleCI reports.

Reply:
I would just as soon keep it named 'test-results' since that's what's produced in the build

Reply:
But I think it might be best to do that Coveralls stuff outside of this PR.

Reply:
to name it 'unit-tests'?

Reply:
I mean the .gitignore file is gone so there shouldn't be a commit issue, so why rename to 'unit-tests'?

Reply:
Ok just remembered one reason – if you take a look at https://github.com/nasa/common-mapping-client/tree/gh-pages/branches/circleci/unit-tests you'll see we're including both test results and coverage in there so "unit-tests" was just some name to group those two things together.